sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1295,8 +1295,14 @@ abstract class AnsiCastSuiteBase extends CastSuiteBase {
       }
     }
   }
-}
 
+  test("SPARK-35720: Support casting of String to timestamp without time zone type") {
+    specialTs.foreach { s =>
```

Comment:
@pingsutw could you test multiple time zone as the default, similar to https://github.com/apache/spark/pull/32878?
Also, we should test input strings like `2021-06-14 15:01:24+08:00` as well, which expects `2021-06-14 15:01:24`.

Reply:
As I mentioned in https://issues.apache.org/jira/browse/SPARK-35720, I don't think we can simply use `stringToTimestamp` here. You will find that when you have more test cases here.

Reply:
@gengliangwang thanks for the review. Will update ASAP.